restapi.go

```diff
@@ -2552,3 +2562,234 @@ func (s *Session) FollowupMessageEdit(appID string, interaction *Interaction, me
+		return
+	}
+
+	err = unmarshal(body, &st)
+	return
+}
+
+// PrivateArchivedThreads returns archived threads in the channel that are private
+// GET /channels/{channel.id}/threads/archived/private
+//
+// before: returns threads before this timestamp
+// limit: optional maximum number of threads to return
+func (s *Session) PrivateArchivedThreads(channelID string, before *time.Time, limit int) (st *ThreadsResponseBody, err error) {
+	uri := EndpointPrivateArchivedThreads(channelID)
+
+	v := url.Values{}
+	if limit > 0 {
+		v.Set("limit", strconv.Itoa(limit))
+	}
+
+	if before != nil {
+		v.Set("before", before.Format(time.RFC3339))
+	}
+
+	if len(v) > 0 {
+		uri += "?" + v.Encode()
+	}
+
+	body, err := s.RequestWithBucketID("GET", uri, nil, EndpointPrivateArchivedThreads(""))
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &st)
+	return
+}
+
+// JoinedPrivateArchivedThreads returns archived threads in the channel
+// that are of type GUILD_PRIVATE_THREAD, and the user has joined
+// GET /channels/{channel.id}/users/@me/threads/archived/private
+//
+// before: returns threads before this timestamp
+// limit: optional maximum number of threads to return
+func (s *Session) JoinedPrivateArchivedThreads(channelID string, before *time.Time, limit int) (st *ThreadsResponseBody, err error) {
+	uri := EndpointJoinedPrivateArchivedThreads(channelID)
+
+	v := url.Values{}
+	if limit > 0 {
+		v.Set("limit", strconv.Itoa(limit))
+	}
+
+	if before != nil {
+		v.Set("before", before.Format(time.RFC3339))
+	}
+
+	if len(v) > 0 {
+		uri += "?" + v.Encode()
+	}
+
+	body, err := s.RequestWithBucketID("GET", uri, nil, EndpointJoinedPrivateArchivedThreads(""))
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &st)
+	return
+}
+
+// ThreadEdit edits the given thread
+// threadID  : The ID of a Thread
+// name       : The new name to assign the thread.
+func (s *Session) ThreadEdit(channelID string, name string) (*Channel, error) {
+	return s.ThreadEditComplex(channelID, &ThreadEditData{
+		Name: name,
+	})
+}
+
+// ThreadEditComplex edits an existing thread, replacing the parameters entirely with ThreadEdit struct
+// threadID  : The ID of a Thread
+// data          : The thread struct to send
+func (s *Session) ThreadEditComplex(threadID string, data *ThreadEditData) (st *Channel, err error) {
+	body, err := s.RequestWithBucketID("PATCH", EndpointChannel(threadID), data, EndpointChannel(threadID))
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &st)
+	return
+}
+
+// ListActiveThreadsResponseBody is the returned data from ListActiveThreads
+type ListActiveThreadsResponseBody struct {
+	Threads []*Channel      `json:"threads"`
+	Members []*ThreadMember `json:"members"`
+}
+
+// ListActiveThreads returns all active threads in the guild,
+// including public and private threads.
+// Threads are ordered by their `id`, in descending order.
+func (s *Session) ListActiveThreads(guildID string) (st *ListActiveThreadsResponseBody, err error) {
```

Comment:
I would just return the threads and members as two return values, rather than having `ListActiveThreadsResponseBody` exported just for this function.